src/org/ods/orchestration/usecase/LeVADocumentUseCase.groovy

```diff
@@ -1744,19 +1744,25 @@ class LeVADocumentUseCase extends DocGenUseCase {
                 // TODO removeme in ODS 4.x
                 version = "${this.project.buildParams.version}-${this.steps.env.BUILD_NUMBER}"
             } else if (this.project.historyForDocumentExists(doc)) {
+                // The document, or a new version of it, has already been created in this same pipeline run.
                 version = this.project.getHistoryForDocument(doc).getVersion()
             } else {
                 def trackingIssues =  this.getDocumentTrackingIssues(doc, ['D', 'Q', 'P'])
                 version = this.jiraUseCase.getLatestDocVersionId(trackingIssues)
-                if (this.project.isWorkInProgress) {
-                    version = (version + 1L).toString() + "-WIP"
-                } else if (docIsCreatedInTheEnvironment(doc)) {
-                    // The document will be generated in this deploy but it is not created yet
+                if (this.project.isWorkInProgress || docIsCreatedInTheEnvironment(doc)) {
```

Comment:
For improving the quality code what do you think if we move all the code inside of this lambda to a method named getVersion(project, doc) and test it in an isolated way?

Reply:
Angel, this sounds like a good idea, but I try to avoid including refactoring in bug-fixing tickets. What if we take note of this for a separate, less-urgent, PR?

Reply:
OK, good idea 👍